src/phoenix/datasets/experiments.py

```diff
@@ -246,18 +174,19 @@ def sync_run_experiment(experiment_input: Tuple[ExampleProtocol, int]) -> Experi
 
         return experiment_payload
 
-    async def async_run_experiment(
-        experiment_input: Tuple[ExampleProtocol, int],
-    ) -> ExperimentPayload:
-        example, repetition_number = experiment_input
+    async def async_run_experiment(test_case: TestCase) -> ExperimentPayload:
+        example, repetition_number = test_case.example, test_case.repetition_number
         start_time = datetime.now()
         output = None
         error = None
         try:
-            if asyncio.iscoroutinefunction(task):
-                output = await task(example.input)
+            # Do not use keyword arguments, which can fail at runtime
+            # even when function obeys protocol.
```

Comment:
💩 Whyyyy? 😢

Reply:
It's really just because the variable name is not really part of the type signature.

In the screenshot below I change the variable names, and function is still protocol compliant, but it just can't be and shouldn't be called with specific keywords.

<img width="605" alt="Screenshot 2024-06-18 at 6 20 12 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/dfb9fd30-de7e-48c0-9b92-fdb879d6b33d">